tony-history-server/app/hadoop/Security.java

```diff
@@ -0,0 +1,44 @@
+package hadoop;
+
+import com.typesafe.config.Config;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import java.io.File;
+import java.io.IOException;
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import play.Logger;
+
+
+/**
+ * The class handles authentication when cluster is security enabled
+ */
+@Singleton
+public class Security {
+  private static final Logger.ALogger LOG = Logger.of(Security.class);
+  private Config appConf;
+
+  @Inject
+  public Security(Config appConf) {
+    this.appConf = appConf;
+    setUpKeytab(Configuration.getHdfsConf());
+  }
+
+  private void setUpKeytab(HdfsConfiguration hdfsConf) {
+    boolean isSecurityEnabled = hdfsConf.get("hadoop.security.authentication").equals("kerberos");
+    if (isSecurityEnabled) {
+      try {
+        UserGroupInformation.setConfiguration(hdfsConf);
+        UserGroupInformation.loginUserFromKeytab(appConf.getString("keytab.user"),
+            appConf.getString("keytab.location"));
```

Comment:
We should define "keytab.location" and "keytab.user" as `static final` class variables instead of inlining them here.

Reply:
I think we can only set the variables (keytab location and keytab user) after the constructor is called, and so we can't declare `final` on the variables. But I agree with the class variable part. We can set directly from `Config` injected from constructor without the `appConf` variable.

Reply:
I just meant defining variables for the string literals `"keytab.user"` and `"keytab.location"`. They could be defined at the top of this class like so:

```
private static final String KEYTAB_USER = "keytab.user";
private static final String KEYTAB_LOCATION = "keytab.location";
```